src/averageAges.js

```diff
@@ -15,6 +15,25 @@
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
+  let newArray = [];
+  let avgAge = 0;
+
+  if (century === undefined) {
+    newArray = people.filter(x => x.sex === 'm');
+
+    avgAge = newArray.reduce((sum, x) => sum + x.died - x.born, 0)
+    / newArray.length;
+
+    return avgAge;
+  }
+
+  newArray = people.filter(x => x.sex === 'm'
+  && Math.ceil(x.died / 100) === century);
```

Comment:
```suggestion
  const men = people.filter(person =>
    century
      ? person.sex === 'm' && Math.ceil(person.died / 100) === century
      : person.sex === 'm');
```
Гораздо короче и понятнее, и этот массив уже можно запускать в reduce. В остальных функциях можно сделать похожим образом